src/main/java/mate/academy/bookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,148 @@
+        return orderResponseDto;
+    }
+
+    @Override
+    @Transactional
+    public List<OrderResponseDto> getOrders(Authentication authentication, Pageable pageable) {
+        User user = (User) authentication.getPrincipal();
+        return orderRepository.findOrdersByUserId(user.getId(), pageable).stream()
+                .map(orderMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public OrderResponseDto updateOrderStatus(Long orderId, OrderStatusRequestDto requestDto) {
+        Order order = orderRepository.findOrderById(orderId).orElseThrow(
+                () -> new EntityNotFoundException("Can't find order by id: " + orderId));
+        order.setStatus(requestDto.getStatus());
+        Order savedOrder = orderRepository.save(order);
+        return orderMapper.toDto(savedOrder);
+    }
+
+    @Override
+    public List<OrderItemResponseDto> getOrderItemsByOrderId(Authentication authentication,
+                                                             Long orderId, Pageable pageable) {
+        Order order = getValidOrder(authentication, orderId);
+        List<OrderItem> orderItems = orderItemRepository.findOrderItemsByOrderId(orderId, pageable);
+        return orderItems.stream()
+                .map(orderItemMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public OrderItemResponseDto getOrderItemByIdAndOrderId(Authentication authentication,
+                                                           Long orderId, Long orderItemId) {
+        Order order = getValidOrder(authentication, orderId);
+        OrderItem orderItem = orderItemRepository.findById(orderItemId).orElseThrow(
+                () -> new EntityNotFoundException("Can't find OrderItem by id: " + orderId));
+        if (!orderItem.getOrder().getId().equals(order.getId())) {
+            throw new OrderException("We don't possess an OrderItem that matches the provided ID "
+                    + orderItemId + " in your Order");
+        }
+        return orderItemMapper.toDto(orderItem);
+    }
+
+    private BigDecimal countTotalOrderPrice(Set<CartItem> cartItems) {
+        return cartItems.stream()
+                .map(c -> c.getBook().getPrice().multiply(BigDecimal.valueOf(c.getQuantity())))
+                .reduce(BigDecimal.ZERO, BigDecimal::add);
+    }
+
+    private Set<OrderItem> fullFillSetOfOrderItems(Order order, Set<CartItem> cartItems) {
+        Set<OrderItem> orderItems = cartItems.stream()
+                .map(c -> fullFillOrderItem(order, c))
+                .collect(Collectors.toSet());
+        orderItemRepository.saveAll(orderItems);
+        return orderItems;
+    }
+
+    private OrderItem fullFillOrderItem(Order order, CartItem cartItem) {
+        OrderItem orderItem = orderItemMapper.toModelFromCartItem(cartItem);
+        orderItem.setOrder(order);
+        return orderItem;
+    }
+
+    private Order getValidOrder(Authentication authentication, Long orderId) {
```

Comment:
Bad naming. The name should represent what the method does. But what is a valid Order?